src/components/UserBadges/index.js

```diff
@@ -20,18 +23,24 @@ type Props = {|
 class UserBadges extends React.Component<Props> {
   @computed
   get _badgesLoader(): LoadObject<Array<Badge>> {
-    return AchievementStore.getAchievementCounters(this.props.userID).map(
-      (achievementCounters: Array<AchievementCounter>): Array<Badge> =>
+    return AchievementStore.getAchievementCounters(this.props.userID)
+      .map((achievementCounters: Array<AchievementCounter>): Array<Badge> =>
         achievementCounters.map(
           (achievementCounter: AchievementCounter): Badge =>
             BADGES_BY_TYPE[achievementCounter.achievementType],
         ),
-    );
+      )
+      .map(
+        (badges: Array<Badge>): Array<Badge> | LoadObject<Array<Badge>> =>
```

Comment:
This should only be called with an array -- never with a LoadObject.

Reply:
why? I want to return LoadObject.empty() instead of empty array, so my LoadComponent could handle the state properly.